src/qibolab/pulses/envelope.py

```diff
@@ -0,0 +1,345 @@
+        return np.array([self.i(samples), self.q(samples)])
+
+
+class Rectangular(BaseEnvelope):
+    """Rectangular envelope."""
+
+    kind: Literal["rectangular"] = "rectangular"
+
+    def i(self, samples: int) -> Waveform:
+        """Generate a rectangular envelope."""
+        return np.ones(samples)
+
+
+class Exponential(BaseEnvelope):
+    r"""Exponential shape, i.e. square pulse with an exponential decay.
+
+    .. math::
+
+        \frac{\exp\left(-\frac{x}{\text{upsilon}}\right) + g \exp\left(-\frac{x}{\text{tau}}\right)}{1 + g}
+    """
+
+    kind: Literal["exponential"] = "exponential"
+
+    tau: float
+    """The decay rate of the first exponential function.
+
+    In units of the interval duration.
+    """
+    upsilon: float
+    """The decay rate of the second exponential function.
+
+    In units of the interval duration.
+    """
+    g: float = 0.1
+    """Weight of the second exponential function."""
+
+    def i(self, samples: int) -> Waveform:
+        """Generate a combination of two exponential decays."""
+        x = np.arange(samples)
+        upsilon = self.upsilon * samples
+        tau = self.tau * samples
+        return (np.exp(-x / upsilon) + self.g * np.exp(-x / tau)) / (1 + self.g)
+
+
+def _samples_sigma(rel_sigma: float, samples: int) -> float:
+    """Convert standard deviation in samples.
+
+    `rel_sigma` is assumed in units of the interval duration, and it is turned in units
+    of samples, by counting the number of samples in the interval.
+    """
+    return rel_sigma * samples
+
+
+class Gaussian(BaseEnvelope):
+    r"""Gaussian pulse shape.
+
+    Args:
+        rel_sigma (float):
+
+    .. math::
+
+        A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}
+    """
+
+    kind: Literal["gaussian"] = "gaussian"
+
+    rel_sigma: float
+    """Relative Gaussian standard deviation.
+
+    In units of the interval duration.
+    """
+
+    def i(self, samples: int) -> Waveform:
+        """Generate a Gaussian window."""
+        return gaussian(samples, _samples_sigma(self.rel_sigma, samples))
+
+
+class GaussianSquare(BaseEnvelope):
+    r"""GaussianSquare pulse shape.
+
+    .. math::
+
+        A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Rise] + Flat + A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Decay]
+    """
+
+    kind: Literal["gaussian_square"] = "gaussian_square"
+
+    rel_sigma: float
+    """Relative Gaussian standard deviation.
+
+    In units of the interval duration.
+    """
+    width: float
+    """Length of the flat portion."""
+
+    def i(self, samples: int) -> Waveform:
+        """Generate a Gaussian envelope, with a flat central window."""
+
+        pulse = np.ones(samples)
+        u, hw = samples / 2, self.width / 2
+        ts = np.arange(samples)
+        tails = (ts < (u - hw)) | ((u + hw) < ts)
+        pulse[tails] = gaussian(len(ts[tails]), _samples_sigma(self.rel_sigma, samples))
+
+        return pulse
+
+
+class Drag(BaseEnvelope):
```

Comment:
```suggestion
class DRAG(BaseEnvelope):
```
I believe this is the PEP recommendation for acronyms.

Reply:
True, but the example in PEP 8 includes acronyms as part of longer identifiers.

Acronyms are always ambiguous, but here in particular would make CamelCase convention the same of SCREAMING_CASE.
That's why I opted for another common convention (i.e. treat acronyms as words).